components/script/document_loader.rs

```diff
@@ -59,7 +60,7 @@ impl LoadBlocker {
 impl Drop for LoadBlocker {
     fn drop(&mut self) {
         if let Some(load) = self.load.take() {
-            self.doc.finish_load(load);
+            self.doc.finish_load(load, CanGc::note());
```

Comment:
This is a really interesting result from this work! I would not have assumed that dropping a LoadBlocker value could cause a JS GC, and it's neat that this PR makes that more clear.